sdk/tables/Azure.Data.Tables/src/TableRestClient.cs

```diff
@@ -76,7 +80,22 @@ public async Task<Response<List<Response>>> SendBatchRequestAsync(HttpMessage me
 
                         if (responses.Length == 1 && responses.Any(r => r.Status >= 400))
                         {
-                            throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(responses[0]).ConfigureAwait(false);
+                            // Batch error messages should be formatted as follows:
+                            // "0:<some error message>"
+                            // where the number prefix is the index of the sub request that failed.
+                            var ex = await _clientDiagnostics.CreateRequestFailedExceptionAsync(responses[0]).ConfigureAwait(false);
+
+                            //Get the failed index
+                            var match = s_entityIndexRegex.Match(ex.Message);
+
+                            if (match.Success && int.TryParse(match.Groups["index"].Value, out int failedEntityIndex))
```

Comment:
Can multiple entities fail?

Reply:
No - the first one that causes a problem fails the whole batch.